### PR TITLE
fix: home accessibility

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

stacked on top of #11199 

There are axe violations on the home page when testing for accessibility

## Changes

* adds flag to let Cypress accessibility tests fail the run
* allows setting aria-label on buttons with no text children
* sets aria-label to be equal to tooltip if it is present and a string
* add story book accessibility testing

I found myself wishing there was an easy way to show that an entire button type was failing contrast check... and well... Storybook will do that

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

running axe dev tools and cypress tests locally

### CODE REVIEW

**Review by clarkus — APPROVED** [review]
Love this!

File: cypress/e2e/a11y.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import { reportA11y } from '../support/accessibility'
 
 describe('a11y', () => {
-    it('home should have no accessibility violations', () => {
+    it.only('home should have no accessibility violations', () => {
```
Comment (benjackwhite) [inline cypress/e2e/a11y.js:4]: If _only_ this wasn't here it would be the perfect PR

**Review by benjackwhite — APPROVED** [review]
Niiiiicee. I wonder if we can get this into an eslint rule... that way its almost no effort to remember and add something when devving

File: cypress/e2e/a11y.js (hunk 1 above)
Comment (pauldambra) [reply cypress/e2e/a11y.js:4]: 20 🦔 pun points :D

**Pull request merged**